std/assembly/internal/typedarray.ts

```diff
@@ -231,3 +231,135 @@ export function EVERY<TArray extends TypedArray<T>, T>(
   }
   return true;
 }
+
+@inline
+export function SET<T extends TypedArray<U>, U extends number, SourceT, SourceU extends number>(
+  target: T,
+  source: SourceT,
+  offset: i32): void {
+  // assert target and source are not null
+  if (isReference<T>(target)) {
+    assert(target != null, "TypeError: Target is null.");
+  } else {
+    assert(false, "TypeError: T is not a reference.");
+  }
+
+  if (isReference<SourceT>(source)) {
+    assert(source != null, "TypeError: Source is null.");
+  } else {
+    assert(false, "TypeError: SourceT is not a reference.");
+  }
+
+  if (isArray<SourceT>(source)) {
+      // check to see if the offsets are in range
+      let sourceLength = source.length;
+      let targetLength = target.length;
+      assert((sourceLength + offset) <= targetLength, "RangeError: Offset is too large.");
+
+      // cache the buffer and the byteOffset
+      let targetBuffer = target.buffer;
+      let targetByteOffset = target.byteOffset;
+
+      // for each source value, write it to the ArrayBuffer
+      for (let i = 0; i < sourceLength; i++) {
+        STORE<U>(
+          targetBuffer,
+          i + offset,
+          <U>unchecked(source[i]),
```

Comment:
use `LOAD` over `source.buffer_`

Reply:
Oh that's right! I think I can normalize a lot of this.

Reply:
This should be resolved with the latest commit.